server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java

```diff
@@ -107,17 +108,17 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    qProfileWsSupport.checkQProfileAdminPermission();
-
-    CreateRequest createRequest = toRequest(request);
     try (DbSession dbSession = dbClient.openSession(false)) {
-      writeProtobuf(doHandle(dbSession, createRequest, request), request, response);
+      OrganizationDto organization = qProfileWsSupport.getOrganizationByKey(dbSession, request.param(PARAM_ORGANIZATION));
+      qProfileWsSupport.checkQProfileAdminPermission(organization);
```

Comment:
I'm not sure it's worth sharing this method checking permissions:

1. the call to `checkLoggedIn` should be the first call in method `handle` so that we don't do any SQL when user is not logged in
2. this leave method `checkQProfileAdminPermission` with a single method call which is hardly useful to factor